### ISSUE TITLE
Solo buttons not functioning

### ISSUE THREAD

**Issue by blackearth2014** [mschack/VCV-Rack-Plugins#7]
![newmixer](https://user-images.githubusercontent.com/7956901/32686432-d550c056-c672-11e7-98c7-e4fd226246d0.png)

I am having an issue with 2x4 and 4x4 solo buttons on sections "B-D" on the 4x4 and section "B" on the 2x4 mixers. The mutes are working fine but the solo button are not muting the other channels when pressed in there sections. I had just download the latest commit release and the problems still exist.

**Comment by mschack** [comment]
Thanks, I will look into it.

**Comment by mschack** [comment]
Fixed in latest.

**Comment by blackearth2014** [comment]
I want to thank you for your fast response and fixes. I come from the days of SSL, Neve and API consoles and I greatly appreciate the the gain stages of your mixers. I have a suggest and a request. If and when you have the time, I think you should make your mixer larger or spread further across the rack a bit. This way you can have the 4 aux sends have four aux returns on the same mixer. Instead of calling it a mixer for 0.5.0 or 1.0, call the "Console".

The Eq's sounds good. If you decide to make the console larger, a bypass per Eq will be nice, a bypass for each aux send/return will be nice, each stereo input channel has its own aux sends as oppose to the whole group section will be nice. My will request if possible if you can implement a bus in and bus out on the master channel. What I mean by this is having the ability to have more of your consoles connect to each other without loosing a channel. One of your mixers can be piped in to another one of your mixers via the bus out of one master channel to the bus in of the main console connected to the audio interface. Propellaheads Reason has this ability via the Mixer14 device.

Sorry for being long winded.

**Comment by mschack** [comment]
Thanks for the feedback.  It is nice to get ideas like this because I am by no means an expert.  When I made a track recently one of the first things I thought was it would be nice to have the AUX sends per channel as opposed per group.  I like to keep things compact (sometimes too compact...) but perhaps you are right in the ability to expand it and add more stuff.  So if I understand;

- ability to chain mixers
- aux per channel as opposed per group ( and not per group as it is now?)
- each aux send with a mix bypass? (in other words send the raw input?)
- EQ bypass
- I am currently adding LED level meters per channel, group and final mix.

**Comment by blackearth2014** [comment]
Yes, Yes, and more Yes. If would keep the 1x4 the same because they will be times when you only want one VCO to be processed 4 different ways before hitting the console. I will save space and routing time. The 2x4  could have only the ability to chain but the 4x4 console should go full Monty.

By the way, your 3 keyboard sequencer is getting nicer and nicer with each release. The 3 OSC sounds good too. The ping pong delay is ridiculously good. Its the best sounding delay in rack thus far. Some documentation will be needed down the road.

One more thing, the aux sends or returns should get some cv features too.

**Comment by mschack** [comment]
Thank you.  Here is a mix I did using most of my stuff.  I have to remind myself to stop developing and use the stuff every once and a while :)  [https://www.youtube.com/watch?v=3WzORPUq4Uc](https://www.youtube.com/watch?v=3WzORPUq4Uc)

**Comment by blackearth2014** [comment]
The audio composition is nice and using your own developments too.

**Comment by blackearth2014** [comment]
MSHack,
I wanted to be clear about about the Eq bypass options. I don't know what algorithms that you are using for code optimizations, but the idea was to only activate the bands that you need to save on processing. Let assume that your Eq's are low shelve, bell and high shelve. If the high shelve is from 4k and above and all I need is a 6k or 8k boost, the other band doesn't need to be processed. Imaging that you have some API equalization code going on and its high quality. 14 channels can really eat up cpu cycles if a mod artist doesn't have signal flow and gain staging background knowledge. He/she wouldn't know why system cycles are spiking.

I hope I made myself clear on this.

**Comment by mschack** [comment]
It is a highpass, bandpass and lowpass filter being mixed together, they are all processed regardless.  Each filter is dependant on the calculation of the other so it is all or nothing.  Now it is possible that I could find a better algorithm, I don't even remember where I got this one from.

**Comment by blackearth2014** [comment]
Autodafe has a fixed filter bank module. Can you borrow some of its code for your eq's?
